controllers/storageoscluster/node-manager_operand.go

```diff
@@ -228,36 +235,18 @@ func (c *NodeManagerOperand) Ensure(ctx context.Context, obj client.Object, owne
 	}
 
 	currState := c.getCurrentState()
-	if len(cluster.Spec.NodeManagerFeatures) > 0 && !currState {
+	if len(cluster.Spec.NodeManagerFeatures) > 0 {
 		err := b.Apply(ctx)
```

Comment:
doesn't the `apply` fail if you already have a resource type? If so, because you removed the `&& !currState` if will fails every time other than the first?

Reply:
There were apply in both branches. Kustomize applies the resource and Kubernetes does what it has to.

Reply:
On the previous branch (main) it would be applied if you had node manager features AND it wasn't applied yet. Now it's always applied

Reply:
https://github.com/storageos/operator/pull/139/files#diff-2f63869e5d0f7c52cdf17acf8a99ce18f81d8f3f52906d249a88c78c61deca6fL245

Reply:
Apply is idempotent in k8s api, so applying the same manifest doesn't hurt.